src/modules/link-tooltip.coffee

```diff
@@ -100,8 +100,17 @@ class LinkTooltip extends Tooltip
 
   _expandRange: (range) ->
     [leaf, offset] = @quill.editor.doc.findLeafAt(range.start, true)
+
+    # Get all adjacent leaves from clicked leaf that share the same link format
+    leaves = [leaf]
+    if leaf?
+      [leaves, offset] = leaf.findAdjacentLeaves(offset, { link: leaf.formats.link })
```

Comment:
I was originally confused why `offset` needed to be passed in here. Now that I've read through the code it makes sense, but it seems like the `offset` related logic is very specific to `_expandRange`.

I think this could be improved by a slight refactor that has `findAdjacentLeaves` return two lists instead, for the previous siblings and the next siblings.

```
[leaf, offset] = @quill.editor.doc.findLeafAt(range.start, true)
start = range.start - offset
end = start + leaf.length

[prevList, nextList] = leaf.findAdjacentLeaves({ link: leaf.formats.link })
prevList.forEach((prev) -> start -= prev.length)
nextList.forEach((next) -> end += next.length)
```
 
In this context, maybe `leaf.findMatchingSiblings` is a better name?

Reply:
I like this. I went with the approach of the Leaf method returning the updated `offset` b/c that seemed to be the convention elsewhere in Quill. But honestly this approach makes more sense to me at first glance.